packages/algob/src/internal/deployer.ts

```diff
@@ -284,6 +348,92 @@ class DeployerBasicMode {
   async ensureCompiled (name: string, force?: boolean, scTmplParams?: SCParams): Promise<ASCCache> {
     return await this.algoOp.ensureCompiled(name, force, scTmplParams);
   }
+
+  /**
+   * Check if ASA info exist for asset,
+   * if it exists check for delete param,
+   * throw error if it is deleted
+   * @param asset asset index or asset name
+   */
+  private checkIfASAExist (asset: string | number): void {
```

Comment:
we need to update the documentation. Be clear that we are searching in checkpoints.

Also:
* why we throw an exception instead of returning `bool`?
* if we want to throw an exception, the the function name should start with `assert...` (eg: assertASAExists)

Reply:
ok will add more details about cases, changing to assert